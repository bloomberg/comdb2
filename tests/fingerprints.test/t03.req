SELECT fingerprint, count, total_cost, total_rows, normalized_sql FROM comdb2_fingerprints WHERE fingerprint NOT IN ('4f16a8ec9db90f803e406659938b2602', '69d1558c950bbb908f4b8df0d63e2cf4', '812bdde365a22a61bcd52bf005990c37') ORDER BY fingerprint;
