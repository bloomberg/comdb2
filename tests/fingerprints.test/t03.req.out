(fingerprint='0a17a3dd4178baa9b98ef60fb54a1992', count=1, total_cost=0, total_rows=2, normalized_sql='SELECT?UNION SELECT?UNION SELECT?;')
(fingerprint='162437a391bff4c53dd483b904a05184', count=1, total_cost=0, total_rows=2, normalized_sql='EXEC PROCEDURE test_fp_close();')
(fingerprint='23476e0d735c4cf482b94c485e3d7127', count=1, total_cost=37, total_rows=7, normalized_sql='EXEC PROCEDURE test_fp();')
(fingerprint='2d4fbab7f6a9c85c75bbb7fb6b43c863', count=1, total_cost=12, total_rows=1, normalized_sql='SELECT sum(x)AS z FROM fp1;')
(fingerprint='3dcb487a727f69d0d6753bf9046d642b', count=1, total_cost=0, total_rows=1, normalized_sql='EXEC PROCEDURE test_fp_emit();')
(fingerprint='5bef07d72bc6a66e49bef817dd3bab9d', count=4, total_cost=0, total_rows=4, normalized_sql='INSERT INTO fp1(x)VALUES(?);')
(fingerprint='5e7c0c2caf9a22cac49643a5a01d7760', count=1, total_cost=13, total_rows=3, normalized_sql='SELECT*FROM fp1 ORDER BY x;')
(fingerprint='732794692d8ab4e61b93a41745a9072a', count=1, total_cost=0, total_rows=1, normalized_sql='CREATE PROCEDURE test_fp_emit VERSION?{
local function main()
  local t, rc = db:exec('SELECT 1')
  db:emit(t)
end};')
(fingerprint='8292b8dcacbbf4b0f44c9864898470af', count=1, total_cost=0, total_rows=1, normalized_sql='CREATE PROCEDURE test_fp VERSION?{
local function exec_fetch_and_emit(sql)
  local rc
  local n, q
  q, rc = db:exec(sql)
  if (rc == 0) then
    n = q:fetch()
    while n do
      db:emit(n)
      n = q:fetch()
    end
  else
    db:emit(db:sqlerror())
  end
end
local function exec_and_nothing(sql)
  local rc
  local q
  q, rc = db:exec(sql)
  if (rc ~= 0) then
    db:emit(db:sqlerror())
  end
end
local function main()
  exec_fetch_and_emit("SELECT 1 AS xyz") -- 1 row
  exec_and_nothing("INSERT INTO fp1(x) VALUES(0)") -- 1 row
  exec_fetch_and_emit("SELECT x AS w FROM fp1 ORDER BY x") -- 2 rows
  exec_fetch_and_emit("SELECT x + x AS y FROM fp1 ORDER BY x") -- 2 rows
  exec_fetch_and_emit("SELECT SUM(x) AS z FROM fp1") -- 1 row
end};')
(fingerprint='84496375cf558c93e22ce29e18ea7187', count=1, total_cost=0, total_rows=0, normalized_sql='CREATE TABLE fp1(x integer);')
(fingerprint='b12bfa0963ad238396f59e2e1e96b218', count=1, total_cost=12, total_rows=1, normalized_sql='SELECT count(*)FROM fp1;')
(fingerprint='b16c596087126535ffc0e991e3477d8a', count=1, total_cost=0, total_rows=1, normalized_sql='SELECT?AS xyz;')
(fingerprint='c7f22188d3a63015bac874fd692fd350', count=1, total_cost=12, total_rows=2, normalized_sql='SELECT x AS w FROM fp1 ORDER BY x;')
(fingerprint='cb377fe3503a8413402ff6fdbf8d9f11', count=1, total_cost=12, total_rows=2, normalized_sql='SELECT x+x AS y FROM fp1 ORDER BY x;')
(fingerprint='d12630393a2ab9bf00df2f53a9355b10', count=1, total_cost=13, total_rows=1, normalized_sql='SELECT x,x+x,sum(x)FROM fp1 ORDER BY x;')
(fingerprint='e2f0f5672b403145980c4535f8ad0253', count=1, total_cost=0, total_rows=1, normalized_sql='CREATE PROCEDURE test_fp_close VERSION?{
local function main()
  local s, rc = db:prepare('SELECT 1 UNION SELECT 2 UNION SELECT 3')
  local row, rc = s:fetch()
  db:emit(row)
  row, rc = s:fetch();
  db:emit(row)
  s:close()
end};')
(fingerprint='ea1c1d9dd89564e809f6501de86061dd', count=1, total_cost=13, total_rows=1, normalized_sql='DELETE FROM fp1 WHERE x=?;')
